fdbserver/CommitProxyServer.actor.cpp

```diff
@@ -1158,16 +1263,21 @@ ACTOR Future<Void> applyMetadataToCommittedTransactions(CommitBatchContext* self
 
 void writeMutation(CommitBatchContext* self, int64_t tenantId, const MutationRef& mutation) {
 	static_assert(TenantInfo::INVALID_TENANT == ENCRYPT_INVALID_DOMAIN_ID);
-	if (!self->pProxyCommitData->isEncryptionEnabled || tenantId == TenantInfo::INVALID_TENANT) {
-		// TODO(yiwu): In raw access mode, use tenant prefix to figure out tenant id for user data
-		bool isRawAccess = tenantId == TenantInfo::INVALID_TENANT && !isSystemKey(mutation.param1) &&
-		                   !(mutation.type == MutationRef::ClearRange && isSystemKey(mutation.param2)) &&
-		                   self->pProxyCommitData->db->get().client.tenantMode == TenantMode::REQUIRED;
-		CODE_PROBE(isRawAccess, "Raw access to tenant key space");
-		self->toCommit.writeTypedMessage(mutation);
-	} else {
+
+	if (self->pProxyCommitData->isEncryptionEnabled) {
+		if (tenantId == TenantInfo::INVALID_TENANT) {
+			std::pair<EncryptCipherDomainName, int64_t> p =
+			    getEncryptDetailsFromMutationRef(self->pProxyCommitData, mutation);
+			CODE_PROBE(true, "Raw access to tenant key space");
```

Comment:
Is this raw access to tenant space or just raw access in general?

Reply:
From encryption perspective, am simplifying things claiming, if `tenantId` (hence, encryption domain) is indeterministic, then consider the mutation as `raw_access`; it is possible the terminology conflicts with existing `raw_access` terminologies, if it does, I will update comments and PROBE accordingly.

Reply:
I agree that without the tenant ID we are using raw access. My concern was with the other part of this text "to tenant key space", which may or may not be true.

Reply:
> to tenant key space", which may or may not be true. 
Agreed, shall update it.